topo_processor/stac/lds_cache.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+import pystac
+
+from topo_processor.util.aws_files import build_s3_path, load_file_content, s3_download
+from topo_processor.util.configuration import lds_cache_local_tmp_folder
+from topo_processor.util.files import decompress_file, is_gzip_file
+
+
+class LdsCache:
+    bucket: str
+    role: str
+
+    def __init__(self, bucket_name: str, role_arn: str):
+        self.bucket = bucket_name
+        self.role = role_arn
+
+    def get_layer(self, layer: str, version_number: str = None) -> str:
+        """Download and return local path to the metadata file for the layer. If no version specified, get the latest version"""
+
+        if not version_number:
+            collection: pystac.Collection = self.get_collection(layer)
+            version_number = self.get_last_version(collection)
+
+        metadata_file_name = self.get_metadata_file_name(layer, version_number)
+        metadata_file_path = lds_cache_local_tmp_folder + "/" + metadata_file_name
+        s3_download(
+            build_s3_path(self.bucket, layer + "/" + metadata_file_name),
+            metadata_file_path,
+            self.role,
+        )
+
+        if os.path.isfile(metadata_file_path):
```

Comment:
What is the best way to handle this? If the `s3_download` run properly (or at least `boto3` not raising any exception), could we have an issue that the file does not exist somehow?